src/cls/rgw/cls_rgw.cc

```diff
@@ -707,10 +707,24 @@ int rgw_bucket_prepare_op(cls_method_context_t hctx, bufferlist *in, bufferlist
   return write_bucket_header(hctx, &header);
 }
 
+static bool obj_name_filter(string& name)
+{
+  int ret = true;
+  if (name.substr(0,10) == MP_MULTI_SUFFIX) {
+    int len = name.size();
+    size_t pos = name.find(MP_META_SUFFIX, len - 5);
```

Comment:
using "find" here is suboptimal.  Use boost::algorithm::ends_with or perhaps name.compare.
Also, -5 is magic - not good to bury knowledge of length of MP_META_SUFFIX like this.